src/Transpiler.ts

```diff
@@ -1019,13 +1018,21 @@ export abstract class LuaTranspiler {
         return parts.join("..");
     }
 
-    public transpileConditionalExpression(node: ts.ConditionalExpression, brackets?: boolean): string {
+    public transpileProtectedConditionalExpression(node: ts.ConditionalExpression): string {
         const condition = this.transpileExpression(node.condition);
         const val1 = this.transpileExpression(node.whenTrue);
         const val2 = this.transpileExpression(node.whenFalse);
+        return `((${condition}) and function() return ${val1}; end or function() return ${val2}; end)()`;
+    }
 
-        return this.transpileLuaLibFunction(LuaLibFeature.Ternary, condition,
-                                            `function() return ${val1} end`, `function() return ${val2} end`);
+    public transpileConditionalExpression(node: ts.ConditionalExpression): string {
+        if (tsHelper.isNonFalsible(this.checker.getTypeAtLocation(node.whenTrue))) {
```

Comment:
What about if strictNullChecks is off? Shouldn't the protected version always be used in this case?

Reply:
Currently we are playing it safe and always protecting non-literal expressions. With strictNullChecks enabled we could also allow non-literal types to be transpiled without protection.
It should be trivial to extend `isNonFalsible` to support strictNullChecks in the future.

Reply:
Update: done.